mesh/pumi.cpp

```diff
@@ -923,223 +943,112 @@ void ParPumiMesh::UpdateMesh(const ParMesh* AdaptedpMesh)
    }
 }
 
-// Transfer a mixed vector-scalar field (i.e. velocity,pressure) and the
-// magnitude of the vector field to use for mesh adaptation.
-void ParPumiMesh::FieldMFEMtoPUMI(apf::Mesh2* apf_mesh,
-                                  ParGridFunction* grid_vel,
-                                  ParGridFunction* grid_pr,
-                                  apf::Field* VelField,
-                                  apf::Field* PrField,
-                                  apf::Field* VelMagField)
+int ParPumiMesh::RotationPUMItoMFEM(apf::Mesh2* apf_mesh,
+                                    apf::MeshEntity* tet,
+                                    int elemId)
 {
-   apf::FieldShape* VelFieldShape = getShape(VelField);
-   int num_nodes = 4 * VelFieldShape->countNodesOn(0) + // Vertex
-                   6 * VelFieldShape->countNodesOn(1) + // Edge
-                   4 * VelFieldShape->countNodesOn(2) + // Triangle
-                   VelFieldShape->countNodesOn(4); // Tetrahedron
-
-   // Define integration points
-   IntegrationRule pumi_nodes(num_nodes);
-   int ip_cnt = 0;
-   apf::Vector3 xi_crd(0.,0.,0.);
-
-   // Create a template of dof holders coordinates in parametric coordinates.
-   // The ordering is taken care of when the field is transferred to PUMI.
-
-   // Dofs on Vertices
-   IntegrationPoint& ip = pumi_nodes.IntPoint(ip_cnt++);
-   double pt_crd[3] = {0., 0., 0.};
-   ip.Set(pt_crd, 3);
-   for (int kk = 0; kk < 3; kk++)
+   MFEM_ASSERT(apf_mesh->getType(tet) == apf::Mesh::TET, "");
+   // get downward vertices of PUMI element
+   apf::Downward vs;
+   int nv = apf_mesh->getDownward(tet,0,vs);
+   int pumi_vid[nv];
```

Comment:
I think only the latest C++ standards, and some compiler extensions, will support variable length arrays.  To be safe side,  we should allocate the `pumi_vid` array or statically size it.

Some discussion here of VLAs in C++:
https://stackoverflow.com/questions/40633344/variable-length-arrays-in-c14

Reply:
fixed in https://github.com/mortezah/mfem/commit/a23760dd801bc2f29245b1b0efc8c4fda43539fe

Also, ex6p runs cleanly under Valgrind.